File: sales-app/src/app/features/model/product_dto.ts
import { CategoryDTO } from './category_dto';

/**
 * DTO que representa la información de un producto
 * expuesta en respuestas HTTP.
 */
export class ProductDTO {
  productCode: string;
  name: string;
  price: number;
  category: CategoryDTO;

  constructor(
    productCode: string,
    name: string,
    price: number,
    category: CategoryDTO
  ) {
    this.productCode = productCode;
    this.name = name;
    this.price = price;
    this.category = category;
  }
}

File: sales/src/main/java/com/gd/sales/product/model/dto/response/ProductDTO.java
package com.gd.sales.product.model.dto.response;

import java.math.BigDecimal;

import com.gd.sales.category.model.dto.response.CategoryDTO;

import lombok.Value;

/**
 * DTO que representa la información de un producto
 * expuesta en respuestas HTTP.
 *
 * @author Luis Monterroso
 * @version 1.0
 * @since 2025-08-02
 */
@Value
public class ProductDTO {
    String productCode;
    String name;
    BigDecimal price;
    CategoryDTO category;
}
